tests/driver.sh

```diff
@@ -465,9 +465,49 @@ try_ 0 << EOF
 #else
 #define A 1
 #endif
+
+#ifndef A
+#define B 1
+#else
+#define B 0
+#endif
 int main()
 {
-    return A;
+    return A + B;
+}
+EOF
+
+# include gaurd test, simulates inclusion of a file named defs.h and global.c
```

Comment:
Shall we add inclusion guard to all headers?

Also, fix typo. Should be "guard"

Reply:
This would be a great enhancement for other editors without having to manually adjust tree-sitter-like plugins to have intellisense functionality. Should I include this modification in other PR or this one?

Reply:
At the moment, shecc has only one header file, `src/defs.h`, so that it is straightforward to enforce inclusion guard.